### PR TITLE
Update supported ansible-version as >=2.14

### PR DESCRIPTION
**Pull request by GomathiselviS**
##### SUMMARY

Dropped support for ansible-version <=2.13 . Updated cloud.common version in galaxy.yml

##### ISSUE TYPE

- Docs Pull Request

##### ADDITIONAL INFORMATION

```paste below

```

### CODE REVIEW

File: galaxy.yml
Comment (gravesm) [inline galaxy.yml:15]: This should either be `>=2.0.4,<4.0.0` or `>=3.0.0,<4.0.0`. I would go with the latter since we're doing a major version upgrade.

Comment (mariolenz) [reply galaxy.yml]: > This should either be `>=2.0.4,<4.0.0` or `>=3.0.0,<4.0.0`. I would go with the latter since we're doing a major version upgrade.

I don't understand this. The dependencies of this collection don't have _anything_ to do with whatever future releases _you_ plan for `cloud.common`.

And: A dependency on `>=3.0.0` doesn't work because the latest release is [2.1.4](https://github.com/ansible-collections/cloud.common/releases/tag/2.1.4) at the moment. This will simply make this collection un-installable IMHO.

Your suggestion doesn't make any sense at all!

Once you've released `cloud.common` 3.0.0, the situation might change. But at the moment, requiring `cloud.common>=3.0.0` would break the collection.

File: galaxy.yml
Comment (mariolenz) [inline galaxy.yml:15]: There's no `cloud.common` 3.0.0 at the moment, so this collection can't depend on it!

Why do you want to define this dependency. Did you test it? HOW? Without a `cloud.common` 3.0.0 available?

@alinabuzachis @jillr

Comment (GomathiselviS) [reply galaxy.yml:15]: We are planning to release cloud.common 3.0.0 soon. This collection will be released after cloud.common 3.0.0 is available. We do test these collections together since vmware_rest has a dependency on cloud.common

Comment (mariolenz) [reply galaxy.yml:15]: > We are planning to release cloud.common 3.0.0 soon. This collection will be released after cloud.common 3.0.0 is available. We do test these collections together since vmware_rest has a dependency on cloud.common

I'm afraid you're not getting the point here. You can't define a dependency on a `cloud common` release that doesn't exist (yet). Release `cloud.common` 3.0.0 and test if if this collection really works with this release. That's a prerequisite for defining it as a requirement.

No matter what you're _planing_ to do, you just can't have a requirement on non-existent collection versions.

Release `cloud.common` 3.0.0 and we can talk again. And, _please_, test if it works.

Comment (jillr) [reply galaxy.yml:15]: Hi @mariolenz We do test this collection with cloud.common and it will be tested as such before this lands.  We're doing a lot of update work in parallel across all of the cloud collections the Ansible team manages right now as the core version update is a requirement for downstream certification.  Where there are cross-collection dependencies that does mean we will have a bit of a timing and coordination issue as we try to pre-stage PRs for initial visual review before other PRs have landed or dependent collections have been released.  My apologies that wasn't clear.  I believe @alinabuzachis is actively working on the cloud.common release, so @GomathiselviS was trying to stage everything here for when that lands.  I think the miscommunication could be that this PR is not marked WIP, I will change that now.

We all have the same objective of ensuring the software that we provide our users will be tested and reliable.  I would appreciate if we could all remain courteous to each other while working together towards that goal.

Comment (mariolenz) [reply galaxy.yml:15]: I'm sorry, I guess I've overreacted a bit. But, you see, dependencies on software versions that don't exist (yet) feels so utterly _wrong_ to me...

Sorry :-/

**Comment by mariolenz** [comment]
ansible-galaxy-importer failed with:

```
ERROR! Failed to resolve the requested dependencies map. Got the candidate cloud.common:2.1.5-dev8 (dependency of vmware.vmware_rest:3.0.0) which didn't satisfy all of the following requirements:
* cloud.common:2.1.5-dev8
* cloud.common:>=3.0.0,<4.0.0
Hint: Pre-releases hosted on Galaxy or Automation Hub are not installed by default unless a specific version is requested. To enable pre-releases globally, use --pre.
Starting galaxy collection install process
Process install dependency map
```

Do we have some kind of [circular dependency](https://github.com/ansible-collections/cloud.common/pull/141#issuecomment-1814897155) here?

**Comment by alinabuzachis** [comment]
recheck

**Comment by mariolenz** [comment]
@alinabuzachis `recheck` doesn't restart the GHA workflows. Let's close and re-open in a few minutes.

**Comment by alinabuzachis** [comment]
> @alinabuzachis `recheck` doesn't restart the GHA workflows. Let's close and re-open in a few minutes.

right, thanks!

**Comment by mariolenz** [comment]
I'm missing the integration tests in ansible/check... I've created #447 to run them with cloud.common 3.0.0.

**Comment by mariolenz** [comment]
Looks like the integration tests succeeded in #447 with cloud.common 3.0.0. What now? Remove the WIP and merge?

**Review by mariolenz — APPROVED** [review]
LGTM

**Comment by GomathiselviS** [comment]
> Looks like the integration tests succeeded in #447 with cloud.common 3.0.0. What now? Remove the WIP and merge?

I'll merge the PR once I receive a couple more approvals. After that, we'll prepare the collection for release.

**Review by hakbailey — APPROVED** [review]

**Pull request merged**
